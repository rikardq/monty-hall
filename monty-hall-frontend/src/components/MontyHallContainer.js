import React, {Component} from 'react';
import axios from "axios/index";
import {DEFAULT_ITERATIONS, STRATEGY_SWITCH} from "../constants";
import MontyHallForm from "./MontyHallForm";
import MontyHallResult from "./MontyHallResult";
import './MontyHallContainer.css';


export default class MontyHallContainer extends Component {

    constructor(props) {
        super(props);
        this.state = {
            loading: false,
            error: false,
            results: []
        };
    }

    startSimulate = async ({strategy = STRATEGY_SWITCH, iterations = DEFAULT_ITERATIONS}) => {
        try {
            this.setState({loading: true});
            const response = await axios.get(`simulate`, {params: {strategy, iterations}});
            if (response.status === 200) {
                this.setState({loading: false, error: false, results: [...this.state.results, response.data]});

            } else {
                this.setState({loading: false, error: true});
            }
        } catch (error) {
            console.error('Error on request ' + error);
            this.setState({loading: false, error: true});
        }
    };

    submitCallback = (params) => {
        // noinspection JSIgnoredPromiseFromCall
        this.startSimulate(params);
    };

    render() {
        return (
            <div className="MontyHallContainer">
                {this.state.error && <h1 className="error">ERROR FROM SERVER :(</h1>}
                <MontyHallForm
                    loading={this.state.loading}
                    submitCallback={this.submitCallback}
                />
                <MontyHallResult
                    results={this.state.results}
                />
            </div>
        );
    }
}