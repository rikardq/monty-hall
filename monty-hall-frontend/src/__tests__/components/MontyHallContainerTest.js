import React from 'react';
import MontyHallContainer from "../../components/MontyHallContainer";
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('renders without crashing', () => {
    const component = shallow(<MontyHallContainer/>);
    expect(component).toMatchSnapshot();
});
