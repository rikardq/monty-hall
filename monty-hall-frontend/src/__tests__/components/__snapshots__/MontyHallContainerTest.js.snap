// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="MontyHallContainer"
>
  <MontyHallForm
    loading={false}
    submitCallback={[Function]}
  />
  <MontyHallResult
    results={Array []}
  />
</div>
`;
